File: 700/728 - Self Dividing Numbers.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ret = new ArrayList<>();
        
        for (int i = left; i <= right; ++i) {
            
            int c = i;
            for (; c > 0; c /= 10) {
                int r = c % 10;
                
                if (r == 0 || i % r != 0) {
                    break;
                }
            }
            
            if (c == 0) {
                ret.add(i);
            }
        }
        
        return ret;
    }
}

File: csharp/728.self-dividing-numbers.cs
/*
 * @lc app=leetcode id=728 lang=csharp
 *
 * [728] Self Dividing Numbers
 */

using System.Linq;

// @lc code=start
public class Solution {
    public IList<int> SelfDividingNumbers(int left, int right) {
        return Enumerable
            .Range(left, right - left + 1)
            .Where(num => {
                int t = num;
                while (t> 0) {
                    int p = t % 10;
                    if (p == 0 || num % p != 0) {
                        return false;
                    }
                    t /= 10;
                }
                return true;
            })
            .ToList();
    }
}
// @lc code=end

